core/ledger/kvledger/kv_ledger.go

```diff
@@ -51,6 +51,7 @@ type kvLedger struct {
 	txmgr                  *txmgr.LockBasedTxMgr
 	historyDB              *history.DB
 	configHistoryRetriever *confighistory.Retriever
+	bookkeepingProvider    bookkeeping.Provider
```

Comment:
we need to get rid of this interface. I will push a PR.

Reply:
https://github.com/hyperledger/fabric/pull/1597 can be merged after this PR. Otherwise, it will create a conflict for this one.